apps/photos/src/services/embeddingService.ts

```diff
@@ -21,26 +22,31 @@ const ENDPOINT = getEndpoint();
 
 const DIFF_LIMIT = 500;
 
-const EMBEDDINGS_TABLE = 'embeddings';
+const EMBEDDINGS_TABLE_V1 = 'embeddings';
+const EMBEDDINGS_TABLE = 'embeddings_v2';
 const EMBEDDING_SYNC_TIME_TABLE = 'embedding_sync_time';
 
-export const getLocalEmbeddings = async () => {
+export const getLocalEmbeddings = async (model?: Model) => {
     const embeddings: Array<Embedding> =
-        (await localForage.getItem<Embedding[]>(EMBEDDINGS_TABLE)) || [];
-    return embeddings;
+        (await localForage.getItem<Embedding[]>(EMBEDDINGS_TABLE)) ?? [];
+    if (!embeddings) {
+        await localForage.removeItem(EMBEDDINGS_TABLE_V1);
+        await localForage.setItem(EMBEDDINGS_TABLE, []);
+        await localForage.setItem(EMBEDDING_SYNC_TIME_TABLE, 0);
+        return [];
+    }
+    if (model) {
+        return embeddings.filter((embedding) => embedding.model === model);
+    } else {
+        return embeddings;
```

Comment:
Why do we return embeddings if the model is empty?

Reply:
so, we needed `allEmbeddings` at some places like, `syncEmbedding` and `cleanupDeletedEmbeddings`, that's why I let not passing any model be the indicator that the caller wants all the embeddings.

Have updated the to have separate `getAllEmbeddings`, as that is safer